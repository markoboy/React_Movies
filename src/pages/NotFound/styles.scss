@import 'essential';

.not-found {
  background: rgb(0,213,217);
  background: linear-gradient(145deg, rgba(0,213,217,1) 0%, rgba(60,60,60,1) 25%, rgba(20,20,20,1) 50%, rgba(0,213,217,1) 100%);
  background-size: 100% 20px;
  animation: move 3s infinite alternate ease-in-out;

  &__text {
    font-size: 5em;
    color: $color-white;
    text-shadow: -6px 1px 8px $color-primary;
    margin: 10px 0 25px;
    position: relative;
    z-index: 10;

    &::before {
      content: '404';
      z-index: 1;
      position: absolute;
      color: $color-white;
      top: 0;
      left: 50%;
      transform: translateX(-50%);
      text-shadow: 5px 1px 10px #00d5d9;
    }
  }
}

@keyframes move {
  from {
    background-position: 0;
    background-size: 100% 20px;
  }

  to {
    background-position: 100vw;
    background-size: 100% 45px;
  }
}