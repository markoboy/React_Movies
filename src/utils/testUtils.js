/* eslint-disable import/no-extraneous-dependencies */
import defaultStore from '@store/index';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export function renderWithStore(
  ui,
  { store = defaultStore, route = '/', ...renderOptions } = {}
) {
  window.history.pushState({}, 'Test page', route);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

export function renderWithRouter(ui, { route = '/' } = {}) {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
}
