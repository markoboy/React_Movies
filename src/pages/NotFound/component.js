import Button from '@components/common/Forms/Button';
import SiteContainer from '@components/features/SiteContainer';
import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

export default function NotFoundComponent() {
  return (
    <SiteContainer classes="background--dark-black justify-content--center text-align--center">
      <h1 className="font--regular">Page Not Found</h1>
      <section className="margin--bottom">
        <div className="not-found">
          <p className="not-found__text">404</p>
        </div>
        <Link to="/">
          <Button classes="btn--outline" tabIndex={-1}>
            Go back to home
          </Button>
        </Link>
      </section>
    </SiteContainer>
  );
}
