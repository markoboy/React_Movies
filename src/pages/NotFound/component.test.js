import React from 'react';

import { renderWithRouter } from '@utils/testUtils';
import NotFound from './component';

describe('NotFound Page', () => {
  it('renders NotFound Page', () => {
    const { container } = renderWithRouter(<NotFound />);

    expect(container).toMatchSnapshot();
  });
});
