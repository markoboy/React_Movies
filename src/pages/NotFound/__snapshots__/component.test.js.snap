// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page renders NotFound Page 1`] = `
<div>
  <main
    class="site-container__main flex flex--column background--dark-black justify-content--center text-align--center"
  >
    <h1
      class="font--regular"
    >
      Page Not Found
    </h1>
    <section
      class="margin--bottom"
    >
      <div
        class="not-found"
      >
        <p
          class="not-found__text"
        >
          404
        </p>
      </div>
      <a
        href="/"
      >
        <button
          class="btn btn--outline"
          tabindex="-1"
          type="button"
        >
          Go back to home
        </button>
      </a>
    </section>
  </main>
</div>
`;
